src/web/api/src/main/java/org/locationtech/geogig/rest/geotools/ExportWebOp.java

```diff
@@ -0,0 +1,158 @@
+        final Context context = super.getContext(request);
+
+        Form options = getRequest().getResourceRef().getQueryAsForm();
+
+        final String srcPath = options.getFirstValue(PATH_PARAM, null);
+        if (srcPath == null) {
+            throw new RuntimeException("\"path\" query parameter must be specified");
+        }
+        final String targetTable= options.getFirstValue(TABLE_PARAM, null);
+        if (targetTable == null) {
+            throw new RuntimeException("\"table\" query parameter must be specified");
+        }
+        final boolean overwrite = Boolean.valueOf(options.getFirstValue(OVERWRITE_PARAM, "false"));
+        final boolean alter = Boolean.valueOf(options.getFirstValue(ALTER_PARAM, "false"));
+        final boolean defaultType = Boolean
+            .valueOf(options.getFirstValue(DEFAULT_TYPE_PARAM, "false"));
+
+        FeatureStoreWrapper featureStoreWrapper;
+        try {
+            featureStoreWrapper = getFeatureStoreWrapper(srcPath, targetTable,options);
+        } catch (IOException ioe) {
+            throw new RuntimeException("Failed to obtain SimpleFeatureStore", ioe);
+        }
+
+        final SimpleFeatureStore featureStore = featureStoreWrapper.getFeatureStore();
+        final ObjectId featureTypeFilterId = featureStoreWrapper.getFeatureTypeFilterId();
+        final File binaryFile = featureStoreWrapper.getBinary();
+
+        // if overwrite is true, remove all the features first
+        if (overwrite) {
+            try {
+                featureStore.removeFeatures(Filter.INCLUDE);
+            } catch (IOException ioe) {
+                throw new RuntimeException("Failed to trucnate table", ioe);
+            }
+        }
+
+        // setup the Export command
+        ExportOp command = context.command(ExportOp.class);
+        if (defaultType) {
+            command.exportDefaultFeatureType();
+        }
+        command.setFeatureStore(featureStoreWrapper.getFeatureStore()).setAlter(alter)
+            .setPath(srcPath).setFilterFeatureTypeId(featureTypeFilterId);
+
+        AsyncContext.AsyncCommand<SimpleFeatureStore> asyncCommand;
+
+        asyncCommand = AsyncContext.get().run(command, getCommandDescription(options));
+
+        MediaType mediaType = variant.getMediaType();
+        return new SimpleFeatureStoreRepresentation(mediaType, asyncCommand, binaryFile);
+
+    }
+
+    /**
+     * Create a SimpleFeatureStore from Form options.
+     *
+     * @param srcPath Path describing which feature to export.
+     * @param targetTable Table name of the destination feature table.
+     * @param options Form option parameters.
+     * @return A wrapper object containing the target SimpleFeatureStore, target SimpleFeatureType
+     * and the ObjectID of the feature filter.
+     * @throws java.io.IOException If a suitable SimpleFeatureStore can't be obtained.
+     */
+    public abstract FeatureStoreWrapper getFeatureStoreWrapper(String srcPath, String targetTable,
```

Comment:
This class exists just to keep all the parts needed for executing the export and getting access to the file to which the export is written.